src/main/java/com/elixr/poc/rest/controller/UserCreationController.java

```diff
@@ -0,0 +1,24 @@
+package com.elixr.poc.rest.controller;
+
+import com.elixr.poc.rest.request.UserRequest;
+import com.elixr.poc.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/application")
+public class UserCreationController {
+    @Autowired
+    UserService userService;
+
+    @PostMapping("/user")
+    public ResponseEntity<?> createUser(@RequestBody UserRequest newUser) {
+        return UserService.builder().build().editResponse(newUser);
```

Comment:
Why are you using the UserService builder to call the editResponse method, even if you have autowired the userService?

Reply:
I would suggest renaming the editResponse method to createUser. I think the purpose of the editResponse method is to create the user right?

Reply:
Actually the purpose of editResponse is to edit the response which we are sending after geting the request.